mdata/write_buffer_test.go

```diff
@@ -0,0 +1,357 @@
+package mdata
+
+import (
+	"fmt"
+	"testing"
+
+	"gopkg.in/raintank/schema.v1"
+)
+
+func testAddAndGet(t *testing.T, reorderWindow uint32, flush func(uint32, float64), testData, expectedData []schema.Point, expectAddFail bool) {
+	b := NewReorderBuffer(reorderWindow, 1, flush)
+	gotFailure := false
+	for _, point := range testData {
+		success := b.Add(point.Ts, point.Val)
+		if !success {
+			gotFailure = true
+		}
+	}
+	if expectAddFail && !gotFailure {
+		t.Fatal("Expected an add to fail, but they all succeeded")
+	}
+	returned := b.Get()
+
+	if len(expectedData) != len(returned) {
```

Comment:
no need to do the len check because the len check will be performed by pointSlicesAreEqual (and also by reflect.DeepEqual)